src/utilities/core-utils.test.ts

```diff
@@ -91,6 +91,145 @@ describe("CoreUtils", () => {
 
```

Comment:
@dtnhinh Looks like you have changes from your other PR in here too. In the future, try to make feature branches on your fork if you're trying to take on multiple issues at once. Thanks!

Reply:
Yeah, the other PR happens to be using the same file so it is combined to one PR like this one unless the first one is merged in first

Reply:
All the comments are addressed in this PR (#68) which will reflex on the other PR (#66) once it is merged in